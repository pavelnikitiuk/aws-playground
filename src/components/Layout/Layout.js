import React from 'react';
import { node } from 'prop-types';
import { Box, Flex, useColorModeValue } from '@chakra-ui/react';
import ThemeModeToggler from '../ThemeModeToggler';

function Layout({ children, ...otherProps }) {
  const background = useColorModeValue('gray.100', 'gray.700');
  const contentBackground = useColorModeValue('white', 'gray.900');
  return (
    <Box bg={background} p={{ md: 10 }}>
      <Flex
        bg={contentBackground}
        border="1px solid rgba(0,0,0,.025)"
        boxShadow="0 1rem 3rem rgba(0,0,0,.175)"
        p={10}
        direction="column"
        maxW={{ xl: '1200px' }}
        m="0 auto"
        {...otherProps}
      >
        <Box position="absolute" top="10" right="10">
          <ThemeModeToggler />
        </Box>
        {children}
      </Flex>
    </Box>
  );
}

Layout.propTypes = {
  children: node.isRequired,
};

export default Layout;
