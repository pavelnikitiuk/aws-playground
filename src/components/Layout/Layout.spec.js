import React from 'react';
import renderer from 'react-test-renderer';
import Layout from './index';

describe('Given the Layout component', () => {
  describe('when it is rendered', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = renderer.create(
        <Layout>Hello</Layout>,
      );
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
