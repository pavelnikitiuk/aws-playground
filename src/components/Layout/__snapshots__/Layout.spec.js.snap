// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the Layout component when it is rendered should match the snapshot 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-kvooi2{background:gray.100;}.css-kvooi2 p{md:10px;}",
      }
    }
    data-emotion="css kvooi2"
  />,
  <div
    className="css-kvooi2"
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-vo4a3{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;background:white;border:1px solid rgba(0,0,0,.025);box-shadow:0 1rem 3rem rgba(0,0,0,.175);padding:10px;margin:0 auto;}.css-vo4a3 maxW{xl:1200px;}",
        }
      }
      data-emotion="css vo4a3"
    />
    <div
      className="css-vo4a3"
    >
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-1opzxad{position:absolute;top:10px;right:10px;}",
          }
        }
        data-emotion="css 1opzxad"
      />
      <div
        className="css-1opzxad"
      >
        <style
          dangerouslySetInnerHTML={
            Object {
              "__html": ".css-198bfjk{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-appearance:none;-moz-appearance:none;-ms-appearance:none;appearance:none;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;position:relative;white-space:nowrap;vertical-align:middle;outline:2px solid transparent;outline-offset:2px;width:auto;padding:0px;}",
            }
          }
          data-emotion="css 198bfjk"
        />
        <button
          aria-label="Toggle theme mode"
          className="chakra-button css-198bfjk"
          type="button"
        >
          <style
            dangerouslySetInnerHTML={
              Object {
                "__html": ".css-onkibi{width:1em;height:1em;display:inline-block;line-height:1em;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;color:currentColor;vertical-align:middle;}",
              }
            }
            data-emotion="css onkibi"
          />
          <svg
            aria-hidden={true}
            className="chakra-icon css-onkibi"
            focusable={false}
            viewBox="0 0 24 24"
          >
            <g
              fill="none"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
            >
              <circle
                cx="12"
                cy="12"
                r="5"
              />
              <path
                d="M12 1v2"
              />
              <path
                d="M12 21v2"
              />
              <path
                d="M4.22 4.22l1.42 1.42"
              />
              <path
                d="M18.36 18.36l1.42 1.42"
              />
              <path
                d="M1 12h2"
              />
              <path
                d="M21 12h2"
              />
              <path
                d="M4.22 19.78l1.42-1.42"
              />
              <path
                d="M18.36 5.64l1.42-1.42"
              />
            </g>
          </svg>
        </button>
      </div>
      Hello
    </div>
  </div>,
]
`;
